rpcs3/rpcs3qt/debugger_frame.cpp

```diff
@@ -532,93 +532,65 @@ void debugger_frame::WritePanels()
 void debugger_frame::ShowGotoAddressDialog()
 {
 	QDialog* diag = new QDialog(this);
-	diag->setWindowTitle(tr("Enter expression"));
+	diag->setWindowTitle(tr("Go To Address"));
 	diag->setModal(true);
 
 	// Panels
 	QVBoxLayout* vbox_panel(new QVBoxLayout());
-	QHBoxLayout* hbox_address_preview_panel(new QHBoxLayout());
 	QHBoxLayout* hbox_expression_input_panel = new QHBoxLayout();
 	QHBoxLayout* hbox_button_panel(new QHBoxLayout());
 
-	// Address preview
-	QLabel* address_preview_label(new QLabel(diag));
-	address_preview_label->setFont(m_mono);
-
 	// Address expression input
 	QLineEdit* expression_input(new QLineEdit(diag));
 	expression_input->setFont(m_mono);
 	expression_input->setMaxLength(18);
-	expression_input->setFixedWidth(190);
 
+	if (auto thread = cpu.lock(); !thread || thread->id_type() != 2)
+	{
+		expression_input->setValidator(new QRegExpValidator(QRegExp("^(0[xX])?0*[a-fA-F0-9]{0,8}$")));
+	}
+	else
+	{
+		expression_input->setValidator(new QRegExpValidator(QRegExp("^(0[xX])?0*[a-fA-F0-9]{0,5}$")));
+	}
+	
 	// Ok/Cancel
 	QPushButton* button_ok = new QPushButton(tr("OK"));
 	QPushButton* button_cancel = new QPushButton(tr("Cancel"));
 
-	hbox_address_preview_panel->addWidget(address_preview_label);
-
 	hbox_expression_input_panel->addWidget(expression_input);
 
 	hbox_button_panel->addWidget(button_ok);
 	hbox_button_panel->addWidget(button_cancel);
 
-	vbox_panel->addLayout(hbox_address_preview_panel);
-	vbox_panel->addSpacing(8);
 	vbox_panel->addLayout(hbox_expression_input_panel);
 	vbox_panel->addSpacing(8);
 	vbox_panel->addLayout(hbox_button_panel);
 
 	diag->setLayout(vbox_panel);
+	expression_input->setFixedWidth(vbox_panel->sizeHint().width());
 
 	const auto cpu = this->cpu.lock();
 
 	if (cpu)
```

Comment:
merge this if/else block with the one above and adjust placeholder text based on input validator and max length

Reply:
Fixed.